### PR TITLE
Refactor primary constructors usage over project

### LINKED ISSUE #233: Refactor all classes to use primary constructor parameters where possible
As part of new .NET class constructor syntax, constructor variables are made available to the entire class (see [Primary Constructors](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/primary-constructors)). As discussed [here](https://github.com/osu-tournament-rating/otr-api/pull/228#discussion_r1548282324), we should be consistent in using this new style and avoid unnecessary `private readonly` instance variables.

Tasks
- [x] Refactor applicable classes to use primary constructor syntax
- [x] Add rule `csharp_style_prefer_primary_constructors` to `.editorconfig`

### PR DESCRIPTION
**Pull request by majorro**
Closes #233 

- Applies primary constructors if not already
- Adds `csharp_style_prefer_primary_constructors` to `.editorconfig`

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -91,6 +91,7 @@ csharp_prefer_static_local_function = true:suggestion
 csharp_prefer_simple_using_statement = false:none
 csharp_style_prefer_switch_expression = true:suggestion
 dotnet_style_readonly_field = true:suggestion
+csharp_style_prefer_primary_constructors = true:suggestion
```
[the remark this answers is not in the document]
Comment (majorro) [reply .editorconfig]: sure

File: API/Utilities/BitLengthAttribute.cs [hunk 2]
```diff
@@ -8,14 +8,8 @@ namespace API.Utilities;
...
 {
-    public int Minimum { get; set; }
-    public int Maximum { get; set; }
-
-    public BitLengthAttribute()
-    {
-        Minimum = 0;
-        Maximum = int.MaxValue;
-    }
+    public int Minimum { get; set; } = 0;
```
[the remark this answers is not in the document]
Comment (majorro) [reply API/Utilities/BitLengthAttribute.cs]: sure

**Comment by majorro** [comment]
> Does this actually close #233? This PR implies there are zero remaining classes which need updating which I am wary of.

Yes, for sure, I looked over all classes after applying the rider automation

**Review by hburn7 — APPROVED** [review]
Have searched for ` _[A-z]+ = ` as a regex to find items like `_field = ` and I don't see any remaining primary constructor conversions.

**Pull request merged**